### ISSUE TITLE
Does a negate or NOT functionality exist?

### ISSUE THREAD

**Issue by veloper** [lucaong/minisearch#109]
### Main Issue

Hi, i was wondering if there is any kind of "negate" operator? (e.g., `ms.search("fruit -apple")`

### Filtering Question / Subquestion
I started top implement this functionality through a filter but it looks like there might be a problem with that approach.
Looking at the source it seems like the `filter` option is only available in the top level `searchOption|options` param passed into `ms.search` or `ms.autoSuggest` methods. e.g.,(`ms.search(this.getQueryCombinations(), searchOptions)` 

So, is my understanding correct when I say that a `filter` can only be applied at the location of the `ms.search` call and it will NOT pay attention to any `filter` option nested within `this.getQueryCombinations()` ?

Thanks for any help here.

**Comment by lucaong** [comment]
Hi @veloper ,
There is no built in negate operator. It would be possible to implement it, but I would want it to have a reasonable API: I prefer to stay away from query languages as string queries (like `”fruit -apple”), because that would break other useful features such as custom tokenizers.

You are right that `filter` is only applied to the final result: as a matter of fact, it’s the same as calling the `filter` Array method on the results (the built in `filter` is mostly there so that one can use it also for auto suggestions, where one does not have access to the result list). How would you want it to work instead? I can think whether it can be implemented in a way that can serve this use-case.

**Comment by lucaong** [comment]
Hi @veloper ,
I implemented a NOT combination in #110 , it works like the following:

```javascript
// Search for documents containing 'nova' but NOT 'bossa'
ms.search({
  combineWith: 'NOT',
  queries: ['nova', 'bossa']
})
```

I want to allow myself a little time to think if this is a good API, and to document it well, but if all is fine I could merge and release this soon. What do you think about it?

**Comment by lucaong** [comment]
Hi @veloper ,
I just merged the pull request mentioned above and released version `v3.2.0` that includes this functionality. It is now possible to address your use-case of searching for documents that match some terms and **do not** match other terms. I am therefore closing this issue, but feel free to comment further if necessary.

**Comment by veloper** [comment]
Thanks @lucaong  - This is a great little lib!
